ops/marketplace-tf/modules/cloud-init/compute-cloud-init.yml

```diff
@@ -0,0 +1,107 @@
+# create the bacalhau user that will run the bacalhau process
+packages:
+  - vim
+
+users:
+  - name: bacalhau
+    groups: docker
+    system: true
+    shell: /usr/sbin/nologin
+
+write_files:
+  # bacalhau config
+  - path: /bacalhau_repo/config.yaml
+    encoding: b64
+    owner: bacalhau:bacalhau
+    permissions: "0600"
+    defer: true
+    content: |
+      ${bacalhau_config_file}
+
+    # tls cert
+  - path: /bacalhau_repo/bacalhau_cert.crt
+    encoding: b64
+    owner: bacalhau:bacalhau
+    permissions: "0400"
+    defer: true
+    content: |
+      ${tls_cert_file}
+
+  # Runs at every boot to ensure disks are mounted. Ensures on instance restarts disks are mounted.
+  - path: /var/lib/cloud/scripts/per-boot/mount-bacalhau-disks.sh
+    permissions: '0755'
+    owner: root:root
+    content: |
+      #!/bin/bash
+      is_mounted() {
+        mountpoint -q "$1"
+      }
+      if ! is_mounted "/bacalhau_repo"; then
+        mount /bacalhau_repo 2>/dev/null || true
+      fi
+      if ! is_mounted "/bacalhau_local_data"; then
+        mount /bacalhau_local_data 2>/dev/null || true
+      fi
+
+  # bacalhau service file to run the bacalhau process
+  - path: /etc/systemd/system/bacalhau.service
+    encoding: b64
+    owner: root:root
+    permissions: "0600"
+    content: |
+      ${bacalhau_service_file}
+
+  # bacalhau install-script to install a custom bacalhau version
+  - path: /etc/install-bacalhau.sh
```

Comment:
/etc probably not the best place for this. /usr/local/sbin's the best place for binaries that need su,or /usr/local/bin if it is intended for wider use.  If it's only used by the installer, then probably less of an issue.